optimum/executorch/modeling.py

```diff
@@ -1139,3 +1139,215 @@ def transcribe(
         self.stats.on_inference_end()
         self.stats.print_report()
         return self.tokenizer.decode(generated_tokens, skip_special_tokens=True)
+
+
+class ExecuTorchModelForMultiModalToText(ExecuTorchModelBase):
+    """
+    An ExecuTorch model for inference of multimodal input to text models using the ExecuTorch Runtime.
+
+    Attributes:
+        auto_model_class (`Type`):
+            Associated Transformers class, `AutoModelForSpeechSeq2Seq`.
+        model (`ExecuTorchModule`):
+            The loaded ExecuTorch model.
+        use_kv_cache (`bool`):
+            Whether key-value caching is enabled. For performance reasons, the exported model is
+            optimized to use a static cache.
+        max_cache_size (`int`):
+            Maximum sequence length supported by the cache.
+        max_batch_size (`int`):
+            Maximum supported batch size.
+        dtype (`str`):
+            Data type of the model parameters.
+        bos_token_id (`int`):
+            Beginning-of-sequence token ID.
+        eos_token_id (`int`):
+            End-of-sequence token ID.
+        vocab_size (`int`):
+            Size of the model vocabulary.
+    """
+
+    # Using general `AutoModel` since it usually routes to the correct model variant and there is no
+    # auto model class that captures both audio and image.
+    auto_model_class = AutoModel
+
+    def __init__(self, models: Dict[str, "ExecuTorchModule"], config: "PretrainedConfig"):
+        super().__init__(models=models, config=config)
```

Comment:
this will register not self.model but self.<decoder> self.<token_embedding> etc, right? https://github.com/huggingface/optimum-executorch/blob/main/optimum/executorch/modeling.py#L98-L103C31. So not sure how this is working

Reply:
There will just be one model.pte, with multiple methods

Reply:
yeah but where does that happen. I couldnt find that in optimum

Reply:
ok i see the changed code in xnnpack recipe